File: python/lcextension/metrics.py
from typing import Dict, List, Any

class Metric(object):
    def __init__(self, sku: str, value: int):
        self.sku: str = sku
        self.value: int = value

    def serialize(self) -> Dict[str, Any]:
        return {
            'sku': self.sku,
            'value': self.value,
        }

class MetricReport(object):
    def __init__(self, metrics: List[Metric], idempotent_key: str):
        self.metrics: List[Metric] = metrics
        self.idempotent_key: str = idempotent_key

    def serialize(self) -> Dict[str, Any]:
        return {
            'idempotent_key': self.idempotent_key,
            'metrics': [metric.serialize() for metric in self.metrics],
        }

File: common/metrics.go
package common

type Metric struct {
	Sku   string `json:"sku" msgpack:"sku"`
	Value uint64 `json:"value" msgpack:"value"`
}

type MetricReport struct {
	IdempotentKey string   `json:"idempotent_key" msgpack:"idempotent_key"`
	Metrics       []Metric `json:"metrics" msgpack:"metrics"`
}
